src/downloader/Downloader.java

```diff
@@ -0,0 +1,67 @@
+package downloader;
+
+import java.io.File;
+
+/**
+ * This class is used to download the traces for the simulator
+ * @author Alex
+ *
+ */
+public class Downloader {
+	private static Downloader INSTANCE = new Downloader();
+	
+	private Downloader(){}
+	
+	public static Downloader getInstance() {
+		return Downloader.INSTANCE;
+	}
+	
+	public void downloadTraces() {
+		
+		if (checkIfTracesExist()) {
+			System.out.println("Trace files exist - Skipping DOWNLOAD STEP");
+		} else {
+			System.out.println("Trace files don't exist - Starting DOWNLOAD STEP");
+			DownloadCore core = new DownloadCore();
+			
+			// Create archives folder
+			File curDir = new File(".");
+			File t = new File(curDir.getAbsolutePath() + File.separator + "ARCHIVES");
+			t.mkdir();
+
+			String traces = curDir.getAbsolutePath();
+			
+			// Download and unzip BEIJING
```

Comment:
The similator runs for only one traces at certain moment, I would do the download for only one scenario/city based on the configuration file when the download is done